core/primitives/src/shard_layout.rs

```diff
@@ -0,0 +1,143 @@
+use crate::borsh::maybestd::io::Cursor;
+use crate::types::{AccountId, NumShards};
+use borsh::{BorshDeserialize, BorshSerialize};
+use byteorder::{LittleEndian, ReadBytesExt};
+use near_primitives_core::hash::hash;
+use near_primitives_core::types::ShardId;
+use serde::{Deserialize, Serialize};
+use std::cmp::Ordering::Greater;
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum ShardLayout {
+    V0(ShardLayoutV0),
+    V1(ShardLayoutV1),
+}
+
+#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct ShardLayoutV0 {
+    /// Map accounts evenly across all shards
+    num_shards: NumShards,
+    /// Parent shards for the shards, useful for constructing states for the shards.
+    /// None for the genesis shard layout
+    parent_shards: Option<Vec<ShardId>>,
```

Comment:
hmm do we actually need this field here? For the new version we just use `ShardLayoutV1`?

Reply:
It's for back compatibility for existing tests. A lot of tests in our code uses more than one shard and use the hash function for account_id_to_shard_id.